### PR TITLE
Fix agent template field and e2e test cases issue.

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: pkg/proxyagent/agent/agent.go [hunk 1]
```diff
@@ -337,7 +337,7 @@ func managedProxyServiceResolverToFilterServiceToExpose(serviceResolvers []proxy
 func convertManagedProxyServiceResolverToService(clusterName string, sr proxyv1alpha1.ManagedProxyServiceResolver) serviceToExpose {
 	return serviceToExpose{
 		Host:         util.GenerateServiceURL(clusterName, sr.Spec.ServiceSelector.ServiceRef.Namespace, sr.Spec.ServiceSelector.ServiceRef.Name),
-		ExternalName: fmt.Sprintf("%s.%s.svc", sr.Spec.ServiceSelector.ServiceRef.Name, sr.Spec.ServiceSelector.ServiceRef.Namespace),
+		ExternalName: fmt.Sprintf("%s.%s.svc.cluster.local", sr.Spec.ServiceSelector.ServiceRef.Name, sr.Spec.ServiceSelector.ServiceRef.Namespace),
```
Comment (qiujian16) [inline pkg/proxyagent/agent/agent.go:340]: why cluster.local?

Comment (xuezhaojun) [reply pkg/proxyagent/agent/agent.go:340]: From the test, it seems that this a necessary suffix. But I need to make a deep invest to answer why... I will bring the result to this thread later.

Comment (xuezhaojun) [reply pkg/proxyagent/agent/agent.go:340]: Found in coreDNS document, says it's not a recursive resolver: https://coredns.io/manual/setups/#recursive-resolver

Comment (xuezhaojun) [reply pkg/proxyagent/agent/agent.go:340]: But in go dns client, it assumes that dns servers in resolv.conf should be a recursive-resolver.  https://github.com/golang/go/blob/6f445a9db55f65e55c5be29d3c506ecf3be37915/src/net/dnsclient_unix.go#L666

**Comment by xuezhaojun** [comment]
@qiujian16 The todo comment is added. Another TODO about security and insecureSkip will be added in cluster-proxy-addon PR.

File: temp/tls.crt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+-----BEGIN CERTIFICATE-----
+MIIEEzCCAvugAwIBAgIIRPA1JEQTgLkwDQYJKoZIhvcNAQELBQAwMDEuMCwGA1UE
```
Comment (qiujian16) [inline temp/tls.crt:2]: nit: we commonly do not put cert/key in github repo. Could them be autogenerated during test?

**Review by qiujian16 — COMMENTED** [review]
/approve

just nit comment, we should not contain cert/key in the PR

File: temp/tls.crt (hunk 2 above)
Comment (xuezhaojun) [reply temp/tls.crt]: Sorry, they are miss upload test files. Deleted.

**Pull request merged**